src/ToolBox/superpmi/superpmi-shared/spmiutil.cpp

```diff
@@ -107,3 +117,110 @@ LPSTR GetCommandLineA()
     return pCmdLine;
 }
 #endif // FEATURE_PAL
+
+void LoadRealJitLib(HMODULE& jitLib, WCHAR* jitLibPath)
+{
+    // Load Library
+    if (jitLib == NULL)
+    {
+        if (jitLibPath == nullptr)
+        {
+            LogError("LoadRealJitLib - No real jit path");
+            return;
+        }
+        jitLib = ::LoadLibraryW(jitLibPath);
+        if (jitLib == NULL)
+        {
+            LogError("LoadRealJitLib - LoadLibrary failed to load '%ws' (0x%08x)", jitLibPath, ::GetLastError());
+            return;
+        }
+    }
+}
+
+void cleanupExecutableName(WCHAR* executableName)
+{
+    WCHAR* quote = nullptr;
+
+    // If there are any quotes in the file name convert them to spaces.
+    while ((quote = wcsstr(executableName, W("\""))) != nullptr)
+    {
+        *quote = W(' ');
+    }
+
+    // Remove any illegal or annoying characters from the file name by converting them to underscores.
+    while ((quote = wcspbrk(executableName, W("=<>:\"/\\|?! *.,"))) != nullptr)
+    {
+        *quote = W('_');
+    }
+}
+
+void generateRandomSuffix(WCHAR* buffer, size_t length)
+{
+    unsigned randNumber = 0;
+#ifdef FEATURE_PAL
+    PAL_Random(&randNumber, sizeof(randNumber));
+#else  // !FEATURE_PAL
+    rand_s(&randNumber);
+#endif // !FEATURE_PAL
+
+    swprintf_s(buffer, length, W("%08X"), randNumber);
+}
+
+WCHAR* getResultFileName(const WCHAR* folderPath, WCHAR* executableName, const WCHAR* extension)
+{
+    cleanupExecutableName(executableName);
+
+    size_t executableNameLength = wcslen(executableName);
+    size_t extensionLength      = wcslen(extension);
+    size_t folderPathLength     = wcslen(folderPath);
+
+    size_t dataFileNameLength = folderPathLength + 1 + executableNameLength + 1 + extensionLength;
+
+    const size_t maxAcceptablePathLength =
+        MAX_PATH - 50; // subtract 50 because excel doesn't like paths longer then 230.
+
+    if (dataFileNameLength > maxAcceptablePathLength)
+    {
+        // The path name is too long; creating the file will fail. This can happen because we use the command line,
+        // which for ngen includes lots of environment variables, for example.
+        // Shorten the executable file name and add a random string to it to avoid collisions.
+
+        const size_t randStringLength = 9;
+
+        size_t lengthToBeDeleted = (dataFileNameLength - maxAcceptablePathLength) + randStringLength;
+
+        if (executableNameLength <= lengthToBeDeleted)
+        {
+            LogError("getResultFileName - path to the output file is too long '%ws\\%ws.%ws(%d)'", folderPath,
+                     executableName, extension, dataFileNameLength);
+            return nullptr;
+        }
+
+        executableNameLength -= lengthToBeDeleted;
+        executableName[executableNameLength] = 0;
+
+        executableNameLength += randStringLength;
+        WCHAR randNumberString[randStringLength + 1];
```

Comment:
and you add one for the null character here, so it's counted twice now?

Reply:
Fixed.